fix: Made images zoomable

dr0pdb opened the pull request:
First step for #1841 

Changes: Made the images zoomable in the LiveFeed. 

Library used: [PhotoView](https://github.com/chrisbanes/PhotoView)

Screenshots for the change: 
![videotogif_2018 01 02_09 55 43](https://user-images.githubusercontent.com/21277837/34474374-fc6ff56c-efa3-11e7-9da6-ed3c5dd082f8.gif)

@iamareebjamal @the-dagger Please review. If it is fine then i will implement it on every image view.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java:
```diff
@@ -70,17 +73,22 @@
+            this.zoomableImageModule = zoomableImageModule;
         } catch (ClassCastException e) {
-            throw new ClassCastException("Activity must implement AdapterCallback.");
+            throw new ClassCastException("Activity must implement Callbacks.");
```
Not good, throwing runtime exceptions based on class casting exception is like using Java like JavaScript. Java is type safe, there should be no need to convert an object in raw type and then check if it is of the correct type. I see that you did not write this implementation, but don't add more to it. There shouldn't be a `ClassCastException` if we expect the right type in the arguments.

Please also open an issue to fix the existing code and removing the exception check

dr0pdb commented:
Changes: Removed the unnecessary check.
@iamareebjamal Please review.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java:
```diff
@@ -70,10 +73,14 @@
     }
 
-    public FeedAdapter(Context context, AdapterCallback adapterCallback, List<FeedItem> feedItems) {
+    public FeedAdapter(Context context, AdapterCallback adapterCallback, ZoomableImageModule zoomableImageModule, List<FeedItem> feedItems) {
```
Instead of adding it to constructor, make it optional and ask it as a setter

iamareebjamal replied:
`ZoomableImageModule` is ambiguous. Use name like `OnImageZoomListener` and the method being `onZoom(String)`

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java:
```diff
@@ -70,10 +73,14 @@
+
+            feedImageView.setOnClickListener(v -> {
+                zoomImage(Utils.parseImageUri(feedItems.get(getPosition()).getFullPicture()));
+            });
```
Once you make it optional, just add a null check here as well

iamareebjamal reviewed (changes requested):
And you have to remove reference to the ZoomListener is onStop of fragment or activity as well or it will create a memory leak

dr0pdb commented:
Changes: Renamed the Listener to ```OnImageZoomListener```, Made it optional by setting it by setter, Removed reference in the onStop().

@iamareebjamal Please review.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java:
```diff
@@ -150,4 +158,18 @@ public void onBindViewHolder(RecyclerViewHolder holder, int position) {
+    }
+
+    private void zoomImage(String imageUri) {
+        onImageZoomListener.onZoom(imageUri);
```
Suppose that the ZoomListener was not null at the time of construction but it is removed later on. Now once the image is clicked, NPE would be thrown.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java:
```diff
@@ -150,4 +158,18 @@ public void onBindViewHolder(RecyclerViewHolder holder, int position) {
+    public void removeOnImageZoomListener() {
+        if (onImageZoomListener != null) {
+            onImageZoomListener = null;
+        }
```
No need to add null check here

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java:
```diff
@@ -874,4 +877,14 @@ public void onMethodCallback(List<CommentItem> commentItems) {
+        ZoomableImageDialogFragment zoomableImageDialogFragment = new ZoomableImageDialogFragment();
+        zoomableImageDialogFragment.setStyle(DialogFragment.STYLE_NO_TITLE,0);
+        Bundle bundle = new Bundle();
+        bundle.putString(ConstantStrings.IMAGE_ZOOM_KEY,imageUri);
```
Add a space after comma
 
In other places as well

iamareebjamal reviewed (changes requested):
Great job. Just a few changes required :smile:

dr0pdb committed:
fix: Made images zoomable

dr0pdb commented:
Changes: Added null check in ```zoomImage```, removed null check in ```removeOnImageZoomListener``` and added commas.
@iamareebjamal Please review.
